great_expectations/expectations/expectation.py

```diff
@@ -2281,6 +2283,27 @@ def _get_final_maturity_level(
 
         return maturity
 
+    def register_save_callback(
+        self, callback: Callable[[Expectation], Expectation]
+    ) -> None:
+        """Provide the Expectation with a callback used to save the Expectation configuration."""
+        self._save_callback = callback
+
+    def transfer_save_callback(self, expectation: Expectation) -> Expectation:
+        """Pass the save callback from this Expectation to another."""
+        expectation.register_save_callback(callback=self._save_callback)
+        return expectation
+
+    @public_api
+    def save(self):
+        """Save this Expectation."""
+        if not self._save_callback:
+            raise RuntimeError(
+                "Expectation must be added to an ExpectationSuite before it can be saved."
+            )
+        expectation = self._save_callback(self)
+        expectation.transfer_save_callback(expectation=self)
```

Comment:
I'm not following what transferring the callback to self does

Reply:
yeah, this is a bit opaque, i had to think a minute to remind myself why this was necessary, which is never a good sign 😄 we'd like to be able to continue iterating and calling save on this expectation object, but once we call `save` the `ExpectationConfiguration` we were pinned to is no longer present in the suite. The `save_callback` function provided by the suite returns the newly created expectation, which includes a new `save_callback`. This method transfers that new callback to our current object, so that the user can continue editing in place.

Reply:
ah, after digesting [James's comment here](https://github.com/great-expectations/great_expectations/pull/9053#discussion_r1417953907) I realize that this method is unnecessary.